### PR TITLE
Add Xeon Phi Coproc Support

### PR DESCRIPTION
**Pull request by ghost**
Knights Corner k1om

### CODE REVIEW

File: libdevcore/Base64.cpp [hunk 1]
```diff
@@ -36,7 +36,11 @@ static inline bool is_base64(byte c)
 	return (isalnum(c) || (c == '+') || (c == '/'));
 }
 
+#ifdef __MIC__
+byte find_base64_char_index(byte c)
```
Comment (bobsummerwill) [inline libdevcore/Base64.cpp:39]: What's the issue with this one?
Would `static byte find_base64_char_index(byte c)` be workable for both, and kill the ifdef?

File: libdevcore/Common.h [hunk 2]
```diff
@@ -37,6 +37,10 @@
 #pragma warning(disable:3682) //call through incomplete class
 #endif
 
+#ifdef __MIC__
```
Comment (bobsummerwill) [inline libdevcore/Common.h]: Please can you do a push/pop pair around only the problematic code and add a comment on exact which code the warning is being masked in?   Is this a boost header issue?

File: libdevcore/Common.h [hunk 3]
```diff
@@ -284,7 +289,11 @@ class Timer
 #define DEV_TIMED_FUNCTION DEV_TIMED_SCOPE(__PRETTY_FUNCTION__)
 #endif
 
+#ifdef __MIC__
+#define DEV_TIMED_ABOVE(S, MS) for (::std::pair<dev::TimerHelper, bool> __eth_t(::dev::TimerHelper(S, MS), true); __eth_t.second; __eth_t.second = false)
```
Comment (bobsummerwill) [inline libdevcore/Common.h]: Is the 765 warning suppression above for this line, or for a similar one in a boost header?
For this local problem, the fix is to introduce a space, like so ... 

```
::std::pair< ::dev::TimerHelper
```

File: libdevcore/Exceptions.h [hunk 4]
```diff
@@ -21,6 +21,10 @@
 
 #pragma once
 
+#ifdef __MIC__
+#pragma warning(disable:811) //exception specification for implicitly declared virtual function
```
Comment (bobsummerwill) [inline libdevcore/Exceptions.h]: Again, please could you detail where the warning being masked is, and also add a push/pop pair?  Thanks!

File: libdevcrypto/CMakeLists.txt [hunk 5]
```diff
@@ -4,7 +4,7 @@ set(EXECUTABLE devcrypto)
 
 file(GLOB HEADERS "*.h")
 add_library(${EXECUTABLE} ${SRC_LIST} ${HEADERS})
-add_definitions(-D_GLIBCXX_USE_CXX11_ABI=1)
+#add_definitions(-D_GLIBCXX_USE_CXX11_ABI=1)
```
Comment (bobsummerwill) [inline libdevcrypto/CMakeLists.txt:7]: What's this one about, please?

**Comment by bobsummerwill** [comment]
PS.   Please could you squash these commits too, and add a more detailed comment for the single commit which is left?   Thanks!

File: libdevcore/Base64.cpp (hunk 1 above)
Comment (ghost) [inline libdevcore/Base64.cpp:39]: I had to remove both "static" and "inline" for Intel Compiler with -mmic.
Will this modification properly function with other build systems?

File: libdevcore/Common.h (hunk 2 above)
Comment (ghost) [inline libdevcore/Common.h]: exact lines are:
libweb3core/libdevcore/Common.h(279)
libweb3core/libdevcore/Common.h(287)

can you please provide an example of the expected syntax for pragmas?

```
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140742.txt:204:/home/libweb3core/libdevcore/Common.h(279): warning #765: use of alternative token "<:" appears to be unintended
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140742.txt:211:/home/libweb3core/libdevcore/Common.h(287): warning #765: use of alternative token "<:" appears to be unintended
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140742.txt:219:/home/libweb3core/libdevcore/Common.h(279): warning #765: use of alternative token "<:" appears to be unintended
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140742.txt:226:/home/libweb3core/libdevcore/Common.h(287): warning #765: use of alternative token "<:" appears to be unintended
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:228:/home/libweb3core/libdevcore/Common.h(279): warning #765: use of alternative token "<:" appears to be unintended
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:234:/home/libweb3core/libdevcore/Common.h(287): warning #765: use of alternative token "<:" appears to be unintended
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:241:/home/libweb3core/libdevcore/Common.h(279): warning #765: use of alternative token "<:" appears to be unintended
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:247:/home/libweb3core/libdevcore/Common.h(287): warning #765: use of alternative token "<:" appears to be unintended
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:253:/home/libweb3core/libdevcore/Common.h(279): warning #765: use of alternative token "<:" appears to be unintended
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:258:/home/libweb3core/libdevcore/Common.h(287): warning #765: use of alternative token "<:" appears to be unintended
/mnt/nfs/System/build_logs/libethereum-build_bot-20160428084701.txt:242:/home/libweb3core/libdevcore/Common.h(279): warning #765: use of alternative token "<:" appears to be unintended
/mnt/nfs/System/build_logs/libethereum-build_bot-20160428084701.txt:248:/home/libweb3core/libdevcore/Common.h(287): warning #765: use of alternative token "<:" appears to be unintended
/mnt/nfs/System/build_logs/libethereum-build_bot-20160428084701.txt:255:/home/libweb3core/libdevcore/Common.h(279): warning #765: use of alternative token "<:" appears to be unintended
/mnt/nfs/System/build_logs/libethereum-build_bot-20160428084701.txt:261:/home/libweb3core/libdevcore/Common.h(287): warning #765: use of alternative token "<:" appears to be unintended
[root@localhost libethereum-buildbot]# 
```

File: libdevcore/Exceptions.h (hunk 4 above)
Comment (ghost) [inline libdevcore/Exceptions.h]: libdevcore/Exceptions.h(39): "dev::Exception::~Exception" 
libdevcore/Exceptions.h(51): "dev::RLPException::~RLPException" 
libdevcore/Exceptions.h(54): "dev::BadCast::~BadCast" 
libdevcore/Exceptions.h(55): "dev::BadRLP::~BadRLP"
libdevcore/Exceptions.h(56): "dev::OversizeRLP::~OversizeRLP" 
libdevcore/Exceptions.h(57): "dev::UndersizeRLP::~UndersizeRLP"
libdevcore/Exceptions.h(59): "dev::BadHexCharacter::~BadHexCharacter" 
libdevcore/Exceptions.h(60): "dev::NoNetworking::~NoNetworking"
libdevcore/Exceptions.h(61): "dev::NoUPnPDevice::~NoUPnPDevice" 
libdevcore/Exceptions.h(62): "dev::RootNotFound::~RootNotFound"
libdevcore/Exceptions.h(63): "dev::BadRoot::~BadRoot" 
libdevcore/Exceptions.h(64): "dev::FileError::~FileError"
libdevcore/Exceptions.h(65): "dev::Overflow::~Overflow" 
libdevcore/Exceptions.h(66): "dev::FailedInvariant::~FailedInvariant"
libdevcore/Exceptions.h(67): "dev::ValueTooLarge::~ValueTooLarge" 
libdevcore/Exceptions.h(69): "dev::InterfaceNotSupported::~InterfaceNotSupported"
libdevcore/Exceptions.h(70): "dev::ExternalFunctionFailure::~ExternalFunctionFailure" 

```
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:263:/home/libweb3core/libdevcore/Exceptions.h(39): warning #811: exception specification for implicitly declared virtual function "dev::Exception::~Exception" is incompatible with that of overridden function "std::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:268:/home/libweb3core/libdevcore/Exceptions.h(39): warning #811: exception specification for implicitly declared virtual function "dev::Exception::~Exception" is incompatible with that of overridden function "boost::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:273:/home/libweb3core/libdevcore/Exceptions.h(51): warning #811: exception specification for implicitly declared virtual function "dev::RLPException::~RLPException" is incompatible with that of overridden function "std::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:278:/home/libweb3core/libdevcore/Exceptions.h(51): warning #811: exception specification for implicitly declared virtual function "dev::RLPException::~RLPException" is incompatible with that of overridden function "boost::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:283:/home/libweb3core/libdevcore/Exceptions.h(54): warning #811: exception specification for implicitly declared virtual function "dev::BadCast::~BadCast" is incompatible with that of overridden function "std::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:288:/home/libweb3core/libdevcore/Exceptions.h(54): warning #811: exception specification for implicitly declared virtual function "dev::BadCast::~BadCast" is incompatible with that of overridden function "boost::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:293:/home/libweb3core/libdevcore/Exceptions.h(55): warning #811: exception specification for implicitly declared virtual function "dev::BadRLP::~BadRLP" is incompatible with that of overridden function "std::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:298:/home/libweb3core/libdevcore/Exceptions.h(55): warning #811: exception specification for implicitly declared virtual function "dev::BadRLP::~BadRLP" is incompatible with that of overridden function "boost::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:303:/home/libweb3core/libdevcore/Exceptions.h(56): warning #811: exception specification for implicitly declared virtual function "dev::OversizeRLP::~OversizeRLP" is incompatible with that of overridden function "std::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:308:/home/libweb3core/libdevcore/Exceptions.h(56): warning #811: exception specification for implicitly declared virtual function "dev::OversizeRLP::~OversizeRLP" is incompatible with that of overridden function "boost::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:313:/home/libweb3core/libdevcore/Exceptions.h(57): warning #811: exception specification for implicitly declared virtual function "dev::UndersizeRLP::~UndersizeRLP" is incompatible with that of overridden function "std::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:318:/home/libweb3core/libdevcore/Exceptions.h(57): warning #811: exception specification for implicitly declared virtual function "dev::UndersizeRLP::~UndersizeRLP" is incompatible with that of overridden function "boost::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:323:/home/libweb3core/libdevcore/Exceptions.h(59): warning #811: exception specification for implicitly declared virtual function "dev::BadHexCharacter::~BadHexCharacter" is incompatible with that of overridden function "std::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:328:/home/libweb3core/libdevcore/Exceptions.h(59): warning #811: exception specification for implicitly declared virtual function "dev::BadHexCharacter::~BadHexCharacter" is incompatible with that of overridden function "boost::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:333:/home/libweb3core/libdevcore/Exceptions.h(60): warning #811: exception specification for implicitly declared virtual function "dev::NoNetworking::~NoNetworking" is incompatible with that of overridden function "std::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:338:/home/libweb3core/libdevcore/Exceptions.h(60): warning #811: exception specification for implicitly declared virtual function "dev::NoNetworking::~NoNetworking" is incompatible with that of overridden function "boost::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:343:/home/libweb3core/libdevcore/Exceptions.h(61): warning #811: exception specification for implicitly declared virtual function "dev::NoUPnPDevice::~NoUPnPDevice" is incompatible with that of overridden function "std::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:348:/home/libweb3core/libdevcore/Exceptions.h(61): warning #811: exception specification for implicitly declared virtual function "dev::NoUPnPDevice::~NoUPnPDevice" is incompatible with that of overridden function "boost::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:353:/home/libweb3core/libdevcore/Exceptions.h(62): warning #811: exception specification for implicitly declared virtual function "dev::RootNotFound::~RootNotFound" is incompatible with that of overridden function "std::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:358:/home/libweb3core/libdevcore/Exceptions.h(62): warning #811: exception specification for implicitly declared virtual function "dev::RootNotFound::~RootNotFound" is incompatible with that of overridden function "boost::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:363:/home/libweb3core/libdevcore/Exceptions.h(63): warning #811: exception specification for implicitly declared virtual function "dev::BadRoot::~BadRoot" is incompatible with that of overridden function "std::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:368:/home/libweb3core/libdevcore/Exceptions.h(63): warning #811: exception specification for implicitly declared virtual function "dev::BadRoot::~BadRoot" is incompatible with that of overridden function "boost::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:373:/home/libweb3core/libdevcore/Exceptions.h(64): warning #811: exception specification for implicitly declared virtual function "dev::FileError::~FileError" is incompatible with that of overridden function "std::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:378:/home/libweb3core/libdevcore/Exceptions.h(64): warning #811: exception specification for implicitly declared virtual function "dev::FileError::~FileError" is incompatible with that of overridden function "boost::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:383:/home/libweb3core/libdevcore/Exceptions.h(65): warning #811: exception specification for implicitly declared virtual function "dev::Overflow::~Overflow" is incompatible with that of overridden function "std::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:388:/home/libweb3core/libdevcore/Exceptions.h(65): warning #811: exception specification for implicitly declared virtual function "dev::Overflow::~Overflow" is incompatible with that of overridden function "boost::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:393:/home/libweb3core/libdevcore/Exceptions.h(66): warning #811: exception specification for implicitly declared virtual function "dev::FailedInvariant::~FailedInvariant" is incompatible with that of overridden function "std::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:398:/home/libweb3core/libdevcore/Exceptions.h(66): warning #811: exception specification for implicitly declared virtual function "dev::FailedInvariant::~FailedInvariant" is incompatible with that of overridden function "boost::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:403:/home/libweb3core/libdevcore/Exceptions.h(67): warning #811: exception specification for implicitly declared virtual function "dev::ValueTooLarge::~ValueTooLarge" is incompatible with that of overridden function "std::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:408:/home/libweb3core/libdevcore/Exceptions.h(67): warning #811: exception specification for implicitly declared virtual function "dev::ValueTooLarge::~ValueTooLarge" is incompatible with that of overridden function "boost::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:413:/home/libweb3core/libdevcore/Exceptions.h(69): warning #811: exception specification for implicitly declared virtual function "dev::InterfaceNotSupported::~InterfaceNotSupported" is incompatible with that of overridden function "std::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:418:/home/libweb3core/libdevcore/Exceptions.h(69): warning #811: exception specification for implicitly declared virtual function "dev::InterfaceNotSupported::~InterfaceNotSupported" is incompatible with that of overridden function "boost::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:423:/home/libweb3core/libdevcore/Exceptions.h(70): warning #811: exception specification for implicitly declared virtual function "dev::ExternalFunctionFailure::~ExternalFunctionFailure" is incompatible with that of overridden function "std::exception::~exception"
/mnt/nfs/System/build_logs/libethereum-build_bot-20160409140902.txt:428:/home/libweb3core/libdevcore/Exceptions.h(70): warning #811: exception specification for implicitly declared virtual function "dev::ExternalFunctionFailure::~ExternalFunctionFailure" is incompatible with that of overridden function "boost::exception::~exception"
[root@localhost libethereum-buildbot]# 

```

File: libdevcrypto/CMakeLists.txt (hunk 5 above)
Comment (ghost) [inline libdevcrypto/CMakeLists.txt:7]: when cross compile with -mmic, use of symbol D_GLIBCXX_USE_CXX11_ABI results in many std::__cxx11::basic_string linker errors.

```
/mnt/nfs/System/build_logs/libethereum-build_bot-20160429061730.txt:449:../libdevcrypto/libdevcrypto.so: undefined reference to `CryptoPP::RandomNumberGenerator::GenerateIntoBufferedTransformation(CryptoPP::BufferedTransformation&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned long long)'
/mnt/nfs/System/build_logs/libethereum-build_bot-20160429061730.txt:451:../libdevcrypto/libdevcrypto.so: undefined reference to `CryptoPP::BufferedTransformation::SetRetrievalChannel(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/mnt/nfs/System/build_logs/libethereum-build_bot-20160429061730.txt:452:../libdevcrypto/libdevcrypto.so: undefined reference to `dev::contentsString(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/mnt/nfs/System/build_logs/libethereum-build_bot-20160429061730.txt:453:../libdevcrypto/libdevcrypto.so: undefined reference to `CryptoPP::RandomPool::GenerateIntoBufferedTransformation(CryptoPP::BufferedTransformation&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned long long)'
/mnt/nfs/System/build_logs/libethereum-build_bot-20160429061730.txt:454:../libdevcrypto/libdevcrypto.so: undefined reference to `CryptoPP::BufferedTransformation::ChannelCreatePutSpace(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned long&)'
/mnt/nfs/System/build_logs/libethereum-build_bot-20160429061730.txt:455:../libdevcrypto/libdevcrypto.so: undefined reference to `dev::writeFile(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, dev::vector_ref<unsigned char const>, bool)'
/mnt/nfs/System/build_logs/libethereum-build_bot-20160429061730.txt:456:../libdevcrypto/libdevcrypto.so: undefined reference to `vtable for CryptoPP::StringSinkTemplate<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >'
/mnt/nfs/System/build_logs/libethereum-build_bot-20160429061730.txt:457:../libdevcrypto/libdevcrypto.so: undefined reference to `dev::fromHex(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, dev::WhenError)'
/mnt/nfs/System/build_logs/libethereum-build_bot-20160429061730.txt:458:../libdevcrypto/libdevcrypto.so: undefined reference to `CryptoPP::BufferedTransformation::ChannelFlush(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, int, bool)'
/mnt/nfs/System/build_logs/libethereum-build_bot-20160429061730.txt:460:../libdevcrypto/libdevcrypto.so: undefined reference to `CryptoPP::BufferedTransformation::ChannelMessageSeriesEnd(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int, bool)'
/mnt/nfs/System/build_logs/libethereum-build_bot-20160429061730.txt:461:../libdevcrypto/libdevcrypto.so: undefined reference to `CryptoPP::BufferedTransformation::ChannelPutModifiable2(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned char*, unsigned long, int, bool)'
/mnt/nfs/System/build_logs/libethereum-build_bot-20160429061730.txt:462:../libdevcrypto/libdevcrypto.so: undefined reference to `dev::isHex(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/mnt/nfs/System/build_logs/libethereum-build_bot-20160429061730.txt:463:../libdevcrypto/libdevcrypto.so: undefined reference to `CryptoPP::BufferedTransformation::ChannelPut2(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned char const*, unsigned long, int, bool)'
/mnt/nfs/System/build_logs/libethereum-build_bot-20160429061730.txt:464:../libdevcrypto/libdevcrypto.so: undefined reference to `dev::fromUUID(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
[root@localhost libethereum-buildbot]# 

```

File: libdevcore/Common.h (hunk 3 above)
Comment (ghost) [inline libdevcore/Common.h]: I will test compilation with alternative spacing.

File: libdevcore/Common.h (hunk 2 above)
Comment (chriseth) [inline libdevcore/Common.h]: This is actually part of c++11: http://en.cppreference.com/w/cpp/language/operator_alternative

File: libdevcore/Common.h (hunk 3 above)
Comment (chriseth) [inline libdevcore/Common.h]: This is actually part of c++11: http://en.cppreference.com/w/cpp/language/operator_alternative

If it is easy to make it compatible for non-compliant compilers, please just introduce spaces.

**Comment by chriseth** [comment]
Please explain what the actual compiler used is and why we should introduce workarounds for non-compliant compilers (assuming that is the reason).

I would say it is ok to introduce some smaller syntax fixes but it is a very bad idea to copy code, especialyl if the compiler is not even part of our test chain.

**Comment by ghost** [comment]
Greetings @chriseth,

I aim to add support for Xeon Phi Co-processors.
Some of the fastest supercomputers in the world are composed of Xeon Phi Co-processors. 
"The No. 1 system, Tianhe-2, and the No. 8 system, Stampede, use Intel Xeon Phi processors to speed up their computational rate. The No. 2 system, Titan, and the No. 6 system, Piz Daint, use NVIDIA GPUs to accelerate computation. "
http://www.top500.org/blog/lists/2015/06/press-release/

Unfortunately, I have not been able to get working binaries for Xeon Phi Co-processors using GCC.
However, the Intel compiler produces valid binaries for the Xeon Phi Co-processors.
The compiler flag "-mmic" compiles binaries into k1om architecture.

I am not interested in long term POW using ethminer.
I'm interested in binary intrinsic optimizations and Design of Experiments.
Additionally, using completely unoptimized code, the Xeon Phi Co-processors model 5110P uses about ~160W of electricity and ~~performs at 15Mhs~~.

EDIT: I misread the log files. the baseline performance is actually 1.2 MHs.

There is room for improvement via implementation of 512bit SIMD instructions for SHA3.

```
[root@localhost libethereum-buildbot]# icpc -V
Intel(R) C++ Intel(R) 64 Compiler for applications running on Intel(R) 64, Version 16.0.2.181 Build 20160204
Copyright (C) 1985-2016 Intel Corporation.  All rights reserved.
[root@localhost libethereum-buildbot]# icc -V
Intel(R) C Intel(R) 64 Compiler for applications running on Intel(R) 64, Version 16.0.2.181 Build 20160204
Copyright (C) 1985-2016 Intel Corporation.  All rights reserved.
```

File: libdevcore/Exceptions.h [hunk 6]
```diff
@@ -32,6 +32,9 @@
 #include "CommonData.h"
 #include "FixedHash.h"
 
+#pragma warning( push )
+#pragma warning(disable:811) //exception specification for implicitly declared virtual function
```
Comment (bobsummerwill) [inline libdevcore/Exceptions.h:36]: What are the implicitly declared virtual functions?    We should just declare them properly, with the same exception specs are the parent class.

Comment (ghost) [inline libdevcore/Exceptions.h:36]: The affect functions are listed below:
libdevcore/Exceptions.h(39): "dev::Exception::~Exception" 
libdevcore/Exceptions.h(51): "dev::RLPException::~RLPException" 
libdevcore/Exceptions.h(54): "dev::BadCast::~BadCast" 
libdevcore/Exceptions.h(55): "dev::BadRLP::~BadRLP"
libdevcore/Exceptions.h(56): "dev::OversizeRLP::~OversizeRLP" 
libdevcore/Exceptions.h(57): "dev::UndersizeRLP::~UndersizeRLP"
libdevcore/Exceptions.h(59): "dev::BadHexCharacter::~BadHexCharacter" 
libdevcore/Exceptions.h(60): "dev::NoNetworking::~NoNetworking"
libdevcore/Exceptions.h(61): "dev::NoUPnPDevice::~NoUPnPDevice" 
libdevcore/Exceptions.h(62): "dev::RootNotFound::~RootNotFound"
libdevcore/Exceptions.h(63): "dev::BadRoot::~BadRoot" 
libdevcore/Exceptions.h(64): "dev::FileError::~FileError"
libdevcore/Exceptions.h(65): "dev::Overflow::~Overflow" 
libdevcore/Exceptions.h(66): "dev::FailedInvariant::~FailedInvariant"
libdevcore/Exceptions.h(67): "dev::ValueTooLarge::~ValueTooLarge" 
libdevcore/Exceptions.h(69): "dev::InterfaceNotSupported::~InterfaceNotSupported"
libdevcore/Exceptions.h(70): "dev::ExternalFunctionFailure::~ExternalFunctionFailure"

Comment (bobsummerwill) [inline libdevcore/Exceptions.h:36]: Aha!   Yes, I see now.

I assume that all the destructors need noexcept decoration?

Please could you try just adding ~Exception() noexcept {} directly, and same for RLPException, and then some nasty  stringing equivalents to the two macros?    You could split the macros over a few lines to make them more readable at the same time.

I think that should work, shouldn't it?

File: libdevcrypto/CMakeLists.txt (hunk 5 above)
Comment (chriseth) [inline libdevcrypto/CMakeLists.txt:7]: `git blame` tells me that this is needed for arch linux.

Comment (bobsummerwill) [inline libdevcrypto/CMakeLists.txt:7]: Yeah - we certainly couldn't just take it out.     Would need some conditionals.

**Pull request closed without merging**